pkg/sql/drop_index.go

```diff
@@ -470,10 +470,10 @@ func (p *planner) dropIndexByName(
 	if err := p.writeSchemaChange(ctx, tableDesc, mutationID, jobDesc); err != nil {
 		return err
 	}
-	p.noticeSender.AppendNotice(pgerror.Noticef(
-		"index %q will be dropped asynchronously and will be complete after the GC TTL",
-		idxName.String(),
-	))
+	p.SendClientNotice(ctx,
+		errors.WithHint(
+			pgerror.Noticef("the data for dropped indices is reclaimed asynchronously"),
+			"The reclamation delay can be customized in the zone configuration for the table."))
```

Comment:
curiousity: when do we use the full stop (`.`) and when do we not?

Reply:
error *messages* are not sentences (no capital and no full stop) because they get inserted into larger strings.

The hints always display in full and as-is and thus can be grammatical.